beacon-chain/db/kv/attestations_test.go

```diff
@@ -0,0 +1,129 @@
+package kv
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db/filters"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+)
+
+func TestStore_AttestationCRUD(t *testing.T) {
+	db := setupDB(t)
+	defer teardownDB(t, db)
+	att := &ethpb.Attestation{
+		Data: &ethpb.AttestationData{
+			Crosslink: &ethpb.Crosslink{
+				Shard:      5,
+				ParentRoot: []byte("parent"),
+				StartEpoch: 1,
+				EndEpoch:   2,
+			},
+		},
+	}
+	ctx := context.Background()
+	if err := db.SaveAttestation(ctx, att); err != nil {
+		t.Fatal(err)
+	}
+	attRoot, err := ssz.HashTreeRoot(att)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !db.HasAttestation(ctx, attRoot) {
+		t.Error("Expected attestation to exist in the db")
+	}
+	retrievedAtt, err := db.Attestation(ctx, attRoot)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !proto.Equal(att, retrievedAtt) {
+		t.Errorf("Wanted %v, received %v", att, retrievedAtt)
+	}
+	if err := db.DeleteAttestation(ctx, attRoot); err != nil {
+		t.Fatal(err)
+	}
+	if db.HasAttestation(ctx, attRoot) {
+		t.Error("Expected attestation to have been deleted from the db")
+	}
+}
+
+func TestStore_Attestations_FiltersCorrectly(t *testing.T) {
```

Comment:
Can this be refactored to a table driven test?
```go
tests := []struct{
   filter *QueryFilter
   expectedNumAtt int
} { ... }
```
It would be easier to maintain if new filters are added. It would also be easier to annotate each test struct to explain the scenario, if necessary.